### ISSUE TITLE
Adblock list update crash

### ISSUE THREAD

**Issue by iccub** [brave/brave-ios#2699]
### Description: 

<img width="598" alt="Zrzut ekranu 2020-07-6 o 12 29 17" src="https://user-images.githubusercontent.com/6950387/86584260-e82be300-bf84-11ea-93b4-d71981af44f3.png">

### Steps to Reproduce 
Hard to reproduce. Looks like it happens on first launch after version/adblock file update

**Actual result:** 

**Expected result:**

**Reproduces how often:** [Easily reproduced, Intermittent Issue]

**Brave Version:** 
1.18, happens on 1.17 too but it's less common

**Device details:** 

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

**Comment by kjozwiak** [comment]
Verification **PASSED** on `iPad Mini 4` running `iOS 12.4.1` using `1.18.1 (20.7.6.15)`

As per https://github.com/brave/brave-ios/pull/2700#issue-444712311, went through the following cases:

* Downloaded `1.17 (20.5.27.21)` and loaded a content heavy website (ensuring it's the active tab)
  * used https://www.deviantart.com, https://giphy.com, https://www.polygon.com, https://www.wired.com, https://www.theverge.com
* Updated to `1.18.1 (20.7.6.15)` and ensured Brave didn't crash and loaded the active tab without issues
* ensured that the shields were blocking ads on the content heavy website
* opened https://zerohedge.com in a new tab and ensured that ads were being blocked
* ensured New Tab Page was correctly updating the stats after upgrading

Basically went through 5 upgrades using each of the above content heavy websites.

--------------

Verification passed on iPhone XR with iOS 13.5 running 1.18(20.07.06.15)
- Verified test plan from #2700 
- Verified no crash on upgrading from older version
- Verified stats updated correctly on upgrade
